lua/oil/view.lua

```diff
@@ -465,6 +465,25 @@ local function render_buffer(bufnr, opts)
   vim.bo[bufnr].modifiable = false
   vim.bo[bufnr].modified = false
   util.set_highlights(bufnr, highlights)
+
+  -- Show original location of trash file as virtual text
+  if adapter.name == "trash" and bufname == "oil-trash:///" then
+    local ns = vim.api.nvim_create_namespace("OilVtext")
```

Comment:
It seems this should show the original path of the thrashed file, but I see nothing.

![nvim_oil_trash_scheme_no_virtual_text](https://github.com/stevearc/oil.nvim/assets/876467/5d7a95c0-bc4c-4f35-9166-725cec50c07a)

This shows Oil in floating mode, but same in normal mode.
The file `gamble.xml` is in trash.

Reply:
This is intentional, but perhaps not intuitive. You're using the Linux version which has support for listing the trash files per-directory. Since you're looking at the trashed files from a specific directory, there's no point to displaying the original path. If you open the root `:Oil --trash /` it shows all trashed files, and the original path for each.

Reply:
Nice, virtual text shows up fine in root trash.
To be honest, it is pretty unintuitive (for me).
What I thought would happen is:
* virtual text is always visible
* listing of thrash files is per-directory recursively downwards

Now that I know how it works, it is definitely usable though.

Reply:
Could you explain more what you mean by "per-directory recursively downwards"?

Reply:
Ok, given the following directory state (trashed files are marked with *):
```
- project
  - fileA
  - fileB
  - fileC*
  - dirA
    - fileA.A
    - fileA.B*
    - fileA.C
    - dirA.D
      - fileA.D.A*
      - fileA.D.B
  - dirB
    - fileB.A
    - fileB.B*
    - fileB.C*
```

`Oil --trash project` shows
```
- fileC -> **virtual text...**
- fileA.B -> **...**
- fileA.D.A -> **...**
- fileB.B -> **...**
- fileB.c -> **...**
```

`Oil --trash dirA` shows
```
- fileA.B -> **...**
- fileA.D.A -> **...**
```

Reply:
I've reworked the display logic. The trash directory will now show files that were trashed from that directory, and also directories that lead to trashed files from subdirs. When you are at the root of a trash scope (the root of a device, or the filesystem root), all trashed files will be visible.
I've also changed it so that trashed files will _always_ have virtual text that displays their original path.